### PR TITLE
feat: make additionalTextEdits be possible to disable

### LINKED ISSUE #233: Golang double import with noNewlineInCmpletion: v:true
My \<cr\> bind 
inoremap <expr> \<cr\>    pumvisible() ? "\<C-y>": "\<cr>"

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
closes #233

### CODE REVIEW

File: autoload/lsp/options.vim [hunk 1]
```diff
@@ -33,6 +33,9 @@ export var lspOptions: dict<any> = {
   ignoreMissingServer: false,
   # Focus on the location list window after LspShowReferences
   keepFocusInReferences: false,
+  # If not apply LSP server-initiated actions after completion
+  # specially if some others plugins had registered to do that
+  noAdditionalTextEdits: false,
```
Comment (yegappan) [inline autoload/lsp/options.vim:38]: Can you rename this to "completionTextEdit" and set the default to true?

File: autoload/lsp/options.vim [hunk 2]
```diff
@@ -33,6 +33,9 @@ export var lspOptions: dict<any> = {
   ignoreMissingServer: false,
   # Focus on the location list window after LspShowReferences
   keepFocusInReferences: false,
+  # If not apply LSP server-initiated actions after completion
+  # specially if some others plugins had registered to do that
```
Comment (yegappan) [inline autoload/lsp/options.vim:37]: Can you change the comment to:
```
# If true, apply the LSP server supplied text edits after a completion. 
# If a snippet plugin is going to apply the text edits, then set this to false to 
# avoid applying the text edits twice.
```

File: doc/lsp.txt [hunk 3]
```diff
@@ -393,11 +393,17 @@ ignoreMissingServer	|Boolean| option.  Do not print a missing language
 						*lsp-opt-keepFocusInReferences*
 keepFocusInReferences	|Boolean| option.  Focus on the location list window
 			after LspShowReferences.
+			By default this is set to false.
+						*lsp-opt-noAdditionalTextEdits*
+noAdditionalTextEdits	|Boolean| option.  If not apply LSP server-initiated
+			actions after completion, specially if some others
+			plugins had registered to do that.
```
Comment (yegappan) [inline doc/lsp.txt:400]: Can you use the text from the above comment and update the help text?

**Review by yegappan — CHANGES_REQUESTED** [review]
Thanks for the fix.

File: autoload/lsp/options.vim (hunk 1 above)
Comment (Shane-XB-Qian) [reply autoload/lsp/options.vim]: ok.

File: autoload/lsp/options.vim (hunk 2 above)
Comment (Shane-XB-Qian) [reply autoload/lsp/options.vim]: done.

File: doc/lsp.txt (hunk 3 above)
Comment (Shane-XB-Qian) [reply doc/lsp.txt]: ok.

**Pull request merged**